File: src/decorator/examples/caretakerrestrictions/SizeRestriction.java
package decorator.examples.caretakerrestrictions;

import factory.examples.caretakers.Caretaker;

public class SizeRestriction extends CaretakerDecorator {

    private final String maxSize; // small, medium, large

    public SizeRestriction(Caretaker wrapped, String maxSize) {
        super(wrapped);
        this.maxSize = maxSize;
    }

    @Override
    public String describe() {
        return super.describe() + " | Tamaño max: " + maxSize;
    }
}

File: src/decorator/examples/caretakerrestrictions/SizeRestriction.js
'use strict';

const CaretakerDecorator = require('./CaretakerDecorator');

class SizeRestriction extends CaretakerDecorator {
  constructor(wrapped, maxSize) {
    super(wrapped);
    this.maxSize = maxSize;
  }
  describe() { return `${super.describe()} | Tamano max: ${this.maxSize}`; }
}

module.exports = SizeRestriction;
